flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/AllocationsInfo.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.adaptive.allocator;
+
+import org.apache.flink.runtime.clusterframework.types.AllocationID;
+import org.apache.flink.runtime.executiongraph.ExecutionGraph;
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.state.KeyGroupRange;
+import org.apache.flink.runtime.state.KeyGroupRangeAssignment;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+class AllocationsInfo {
```

Comment:
I'm wondering whether we could incorporate size estimates into the output of this class so we only have a single data structure that we need to pass around 🤔

Also, looking at how `#getAllocations` is used, what we want is a data structure that is indexed by JobVertexID, so we don't have to iterate over all allocations all over again.

Something along the lines of:

```
static class AllocatedState {
  private final AllocationId allocationId;
  private final KeyGroupRange keyGroupRange;
  private final long keyGroupSizeEstimateInBytes;
}

Collection<AllocatedState> getAllocatedStates(JobVertexID);
```

Reply:
I agree, good points :+1:

Reply:
Implemented in 6d4971f8f646409ff09173f7dfe1d5245b95706f and 5fe039b4cb507fcfcc88906c40d86561fc9e8ed5

Reply:
nit: it would be great to make the naming consistent with `JobInformation`; e.g. calling the outer class `JobAllocationInformation` and the inner one `VertexAllocationInformation`. WDYT?

Reply:
Renamed in a4ae3d608e449dd448861dbd1d0fb1f129d5824b.